tests/vmi_multus_test.go

```diff
@@ -665,6 +671,28 @@ var _ = Describe("SRIOV", func() {
 			return
 		}
 
+		createSriovVMI := func(networksNames []string) (vmi *v1.VirtualMachineInstance) {
+			SriovVMI := getSriovVmi(networksNames)
```

Comment:
I am not sure this should be in this PR scope, but since you already create new function that create VMI's could you adjust it to return error instead of handling it in place with `Expect`?
I know that there are places that you cant do, it because it depends on functions that already handle the error with Expecter (e.g.: WaitUntilVMIReady and WaitAgentConnected)
Lets try to adopt the principal of using Expect is only at the top level of the test and not inside helper functions, at least for this file tests.
@AlonaKaplan  @EdDev

Reply:
Yes, not using asserts in helpers is better and I would vote for that.
The error should include all needed info that is available, so there may be a need to encapsulate it within a wrapping error.

Reply:
NP. after this will get merge I will make another PR to change that.

Reply:
I think that handling the error inside the method is fine. I cannot think of a use case that a caller of the method would like to have another spacial handling for error generating the mac.
But up to you, if you want to change it in a follow-up PR I won't block it.

Reply:
> I think that handling the error inside the method is fine. I cannot think of a use case that a caller of the method would like to have another special handling for error generating the mac.
> But up to you, if you want to change it in a follow-up PR I won't block it.

Asserting in helpers is bad because you want to see what failed in the test level and not down the stack (otherwise it is harder to identify where exactly in the test it failed). Gomega gives you an option to an "offset" version of the assert, but it is both ugly and unmaintainable, because helpers may be called from different levels, you cannot control it for the long run (see [here](http://onsi.github.io/gomega/#making-assertions-in-helper-functions)).
In addition, many times it is helpful to gather info on the way up the stack in case of an error, something you cannot do if you panic/assert down there.

We should avoid asserting in helpers as much as possible.